src/filters/java.js

```diff
@@ -2,15 +2,18 @@ import { parseToEntries, stringifyFromEntries } from '@js.properties/properties'
 
 export class JavaPropertiesFilter {
     async parseResource({ resource }) {
-        const parsedResource = parseToEntries(resource, { sep: true, eol: true, all: true, original: true });
+        const parsedResource = parseToEntries(resource, { sep: true, eol: true, all: true, original: true, location: true });
         const segments = [];
         let previousComments = [];
         for (const e of parsedResource) {
             if (e.key && e.sep.trim() === '=') {
+                const location = {startLine: e.location.start.line, endLine: e.location.end.line}
                 const seg = {
                     sid: e.key,
                     str: e.element,
+                    location
```

Comment:
at some point i'd like to tighten/enforce schemas. since this new `location` property is not consistently populated and is not really needed for translation i'm not sure if it should be a top-level segment property or it should be part of a schema-less generic container. for now it's ok but may want to refactor in the future.
btw, since this wasn't added to `schemas.js` it won't be included in TUs, which is probably fine.